modules/mailchimp_automations/includes/mailchimp_automations.admin.inc

```diff
@@ -0,0 +1,335 @@
+      $fields = mailchimp_automations_email_fieldmap_options($form_entity_type, $form_bundle);
+      $form['drupal_entity']['email_property'] = array(
+        '#title' => t('Email Property'),
+        '#type' => 'select',
+        '#required' => TRUE,
+        '#description' => t('Select the field which contains the email address'),
+        '#options' => $fields,
+        '#default_value' => $entitynotnull ? $mailchimp_automations_entity->email_property : 0,
+        '#maxlength' => 127,
+        '#weight' => -6,
+        '#ajax' => array(
+          'callback' => 'mailchimp_automations_mapping_form_callback',
+          'wrapper' => 'mailchimp-automation-drupal-entity',
+        ),
+      );
+
+      $form_email_property = & $form_state['values']['email_property'];
+      if (!$form_email_property && $entitynotnull) {
+        $form_email_property = $mailchimp_automations_entity->email_property;
+      }
+      if ($form_email_property) {
+        $mc_lists = mailchimp_get_lists();
+        $options = array(t('-- Select --'));
+        foreach ($mc_lists as $list_id => $list) {
+          $options[$list_id] = $list->name;
+        }
+        if (!empty($options)) {
+          $form['drupal_entity']['list_id'] = array(
+            '#title' => t('MailChimp List'),
+            '#type' => 'select',
+            '#required' => TRUE,
+            '#description' => t('Select the list that you created in MailChimp.'),
+            '#options' => $options,
+            '#default_value' => $entitynotnull ? $mailchimp_automations_entity->list_id : 0,
+            '#maxlength' => 127,
+            '#weight' => -5,
+            '#ajax' => array(
+              'callback' => 'mailchimp_automations_mapping_form_callback',
+              'wrapper' => 'mailchimp-automation-drupal-entity',
+            ),
+          );
+        }
+        else {
+          $msg = t('You must add at least one list in MailChimp for this functionality to work.');
+          drupal_set_message($msg, 'error', FALSE);
+        }
+
+        $form_list_id = & $form_state['values']['list_id'];
+        if (!$form_list_id && $entitynotnull) {
+          $form_list_id = $mailchimp_automations_entity->list_id;
+        }
+        if ($form_list_id) {
+          $automations = mailchimp_automations_get_automations();
+          if (!empty($automations)) {
+            $automation_options = mailchimp_automations_options_list($automations);
+            $form['drupal_entity']['workflow_id'] = array(
+              '#title' => t('Workflow ID'),
+              '#type' => 'select',
+              '#required' => TRUE,
+              '#description' => t('Select the workflow automation that you created in MailChimp. Untitled workflows are not included.'),
+              '#options' => $automation_options,
+              '#default_value' => $entitynotnull ? $mailchimp_automations_entity->workflow_id : 0,
+              '#maxlength' => 127,
+              '#weight' => -5,
```

Comment:
This `#weight` is the same as `$form['drupal_entity']['list_id']['#weight']`, which makes the Workflow ID field appear above the MailChimp List field. It's a really minor thing, but it's weird to have a field appear above another when loaded via AJAX. Can this be changed to `-4` and `$form['drupal_entity']['workflow_email_id']` be changed to `-3`?